lib/compress/zstd_opt.c

```diff
@@ -818,6 +968,17 @@ ZSTD_compressBlock_opt_generic(ZSTD_matchState_t* ms,
     ZSTD_match_t* const matches = optStatePtr->matchTable;
     ZSTD_optimal_t lastSequence;
 
+    U32 ldmStartPosInBlock = 0;
+    U32 ldmEndPosInBlock = 0;
+    U32 ldmOffset = 0;
+
+    /* Get first match from ldm seq store if long mode is enabled */
+    if (ms->ldmSeqStore.size > 0 && ms->ldmSeqStore.pos < ms->ldmSeqStore.size) {
+        ZSTD_opt_getNextLdmAndUpdateSeqStore(&ms->ldmSeqStore, &ldmStartPosInBlock,
+                                             &ldmEndPosInBlock, &ldmOffset,
+                                             (U32)(ip-istart), (U32)(iend-ip));
+    }
```

Comment:
I think these checks should be moved inside of the LDM functions, so they aren't duplicated. Another benefit is that it will minimize the amount of LDM logic that the optimal parser has, just these 4 functions calls.

E.g. Here you check `size > 0` and below you check `size != 0`. They are equivalent, but it shows that differences can crop up.